spec/consensus/light-client.md

```diff
@@ -1,318 +1,577 @@
+    // return an error if there are no trusted state for the given height
+    // if height = 0, return the latest trusted state
+    func get(store Store, height int64) (TrustedState, error)
+```
 
-_Remark_: This failure model might change to a hybrid version that takes heights into account in the future.
+### Failure Model
 
-The specification in this document considers an implementation of the lite client under this assumption. Issues like _counter-factual signing_ and _fork accountability_ and _evidence submission_ are mechanisms that justify this assumption by incentivizing validators to follow the protocol.
-If they don't, and we have more that 1/3 faults, safety may be violated. Our approach then is to _detect_ these cases (after the fact), and take suitable repair actions (automatic and social). This is discussed in an upcoming document on "Fork accountability". (These safety violations include the lite client wrongly trusting a header, a fork in the blockchain, etc.)
+The lite client specification is defined with respect to the following failure model: If a block `b` is generated
+at time `Time` (and this time is stored in the block), then a set of validators that hold more than 2/3 of the voting
+power in `validators(b.Header.NextValidatorsHash)` is correct until time `b.Header.Time + TRUSTED_PERIOD`.
 
-## Lite Client Trusting Spec
+*Assumption*: "correct" is defined w.r.t. realtime (some Newtonian global notion of time, i.e., wall time),
+while `Header.Time` corresponds to the [BFT time](bft-time.md). In this note, we assume that clocks of correct processes
+are synchronized (for example using NTP), and therefore there is bounded clock drift (CLOCK_DRIFT) between local clocks and
+BFT time. More precisely, for every correct process p and every header (correctly generated by the Tendermint consensus)
+time (BFT time) the following inequality holds: `Header.Time < now + CLOCK_DRIFT`.
 
-The lite client communicates with a full node and learns new headers. The goal is to locally decide whether to trust a header. Our implementation needs to ensure the following two properties:
+Furthermore, we assume that trust period is (several) order of magnitude bigger than clock drift (`TRUST_PERIOD >> CLOCK_DRIFT`),
+as clock drift (using NTP) is in the order of milliseconds and `TRUSTED_PERIOD` is in the order of weeks.
 
-- Lite Client Completeness: If header _h_ was correctly generated by an instance of Tendermint consensus (and its age is less than the trusting period), then the lite client should eventually set _trust(h)_ to true.
+*Remark*: This failure model might change to a hybrid version that takes heights into account in the future.
 
-- Lite Client Accuracy: If header _h_ was _not generated_ by an instance of Tendermint consensus, then the lite client should never set _trust(h)_ to true.
+The specification in this document considers an implementation of the lite client under the Failure Model defined above. Issues
```

Comment:
We should probably mention in this paragraph or as a Remark that we expect there to be some period during which incorrect validators can be punished, call it the PUNISHMENT_PERIOD and that `TRUSTED_PERIOD < PUNISHMENT_PERIOD`. We can even be more precise somewhere with "how much less". We should also note that PUNISHMENT_PERIOD is often referred to as an unbonding period due to the "bonding" mechanism in modern proof of stake ...